src/spyglass/decoding/v0/clusterless.py

```diff
@@ -0,0 +1,862 @@
+    ):
+        """Populate mark indicators for all units in the given spike sorting selection.
+
+        This function is a way to do several pipeline steps at once. It will:
+        1. Populate the SpikeSortingSelection table
+        2. Populate the SpikeSorting table
+        3. Populate the Curation table
+        4. Populate the CuratedSpikeSortingSelection table
+        5. Populate UnitMarks
+        6. Compute UnitMarksIndicator for each position epoch
+
+        Parameters
+        ----------
+        spikesorting_selection_keys : dict
+        mark_param_name : str, optional
+        position_info_param_name : str, optional
+        """
+        spikesorting_selection_keys = deepcopy(spikesorting_selection_keys)
+        # Populate spike sorting
+        SpikeSortingSelection().insert(
+            spikesorting_selection_keys,
+            skip_duplicates=True,
+        )
+        SpikeSorting.populate(spikesorting_selection_keys)
+
+        # Skip any curation
+        curation_keys = [
+            Curation.insert_curation(key) for key in spikesorting_selection_keys
+        ]
+
+        CuratedSpikeSortingSelection().insert(
+            curation_keys, skip_duplicates=True
+        )
+        CuratedSpikeSorting.populate(
+            CuratedSpikeSortingSelection() & curation_keys
+        )
+
+        # Populate marks
+        mark_parameters_keys = pd.DataFrame(CuratedSpikeSorting & curation_keys)
+        mark_parameters_keys["mark_param_name"] = mark_param_name
+        mark_parameters_keys = mark_parameters_keys.loc[
+            :, UnitMarkParameters.primary_key
+        ].to_dict("records")
+        UnitMarkParameters().insert(mark_parameters_keys, skip_duplicates=True)
+        UnitMarks.populate(UnitMarkParameters & mark_parameters_keys)
+
+        # Compute mark indicators for each position epoch
+        nwb_file_name = spikesorting_selection_keys[0]["nwb_file_name"]
+        position_interval_names = (
+            IntervalPositionInfo()
+            & {
+                "nwb_file_name": nwb_file_name,
+                "position_info_param_name": position_info_param_name,
+            }
+        ).fetch("interval_list_name")
+
+        for interval_name in tqdm(position_interval_names):
+            position_interval = IntervalList & {
+                "nwb_file_name": nwb_file_name,
+                "interval_list_name": interval_name,
+            }
+
+            marks_selection = (
+                UnitMarks & mark_parameters_keys
+            ) * position_interval
+            marks_selection = (
+                pd.DataFrame(marks_selection)
+                .loc[:, marks_selection.primary_key]
+                .to_dict("records")
+            )
+            UnitMarksIndicatorSelection.insert(
+                marks_selection, skip_duplicates=True
+            )
+            UnitMarksIndicator.populate(marks_selection)
+
+except ImportError as e:
+    logger.warning(e)
```

Comment:
Ah, missed this when commenting above. I would migrate up to wrap specific packages